ExtractUserDataActivity/handler.ts

```diff
@@ -379,32 +388,66 @@ export const queryAllUserData = (
     .chain(profile =>
       sequenceS(taskEither)({
         // queries all messages for the user
-        messages: fromQueryEither<ReadonlyArray<RetrievedMessageWithContent>>(
-          () => iteratorToArray(messageModel.findMessages(fiscalCode)),
-          "findMessages"
-        ),
+        messages: tryCatch(
+          () =>
+            asyncIterableToArray(
+              messageModel.getQueryIterator({
+                parameters: [
+                  {
+                    name: "@fiscalCode",
+                    value: fiscalCode
+                  }
+                ],
+                query: `SELECT * FROM m WHERE m.fiscalCode = @fiscalCode`
+              })
+            ),
+          toCosmosErrorResponse
+        )
+          .map(flatten)
+          .mapLeft(_ =>
+            ActivityResultQueryFailure.encode({
+              kind: "QUERY_FAILURE",
+              query: "findMessages",
+              reason: `${_.kind}, ${getMessageFromCosmosErrors(_)}`
+            })
+          )
+          .foldTaskEither(
+            _ => fromLeft(_),
+            maybeMessages =>
+              maybeMessages.some(isLeft)
+                ? fromLeft(
+                    ActivityResultQueryFailure.encode({
+                      kind: "QUERY_FAILURE",
+                      query: "findMessages",
+                      reason: "Some messages cannot be decoded"
+                    })
+                  )
+                : rights(maybeMessages).length > 0
```

Comment:
This code is used to check if the array of rights elements contains at least 1 element. If not we have to return an error because no valid messages are present. 
I've called it maybeMessages because array elements are `Either<Errors, RetrievedMessage>`, maybe I've just to call it `errorOrMessages`

Reply:
since the pattern errorOr* is usually tied to an Either, I'd just call it `items` or `results` here, then, after `rigths()` it becomes `messages`. I'm not sure that we want to return an error if no message if found (is that the behavior of the original code ?) but rather just map the empty array

Reply:
cc: @balanza

Reply:
naming fixes in [#e6e2575](https://github.com/pagopa/io-functions-admin/pull/65/commits/e6e2575ed93102aa692f1732c0e859b710b57a7d)

Reply:
> since the pattern errorOr* is usually tied to an Either, I'd just call it `items` or `results` here, then, after `rigths()` it becomes `messages`. I'm not sure that we want to return an error if no message if found (is that the behavior of the original code ?) but rather just map the empty array

fixed in [#4905ffb](https://github.com/pagopa/io-functions-admin/pull/65/commits/4905ffb1cc4f220247f626385dd481a3ad3962b6)

Reply:
The array you check the length of, has only `right` elements, hasn't it?

If so, I think there's no error in having an empty array (that basically means the user has no message - weird, but possibile)